platform/linuxbsd/joypad_linux.cpp

```diff
@@ -448,104 +440,124 @@ Input::JoyAxisValue JoypadLinux::axis_correct(const input_absinfo *p_abs, int p_
 	return jx;
 }
 
-void JoypadLinux::process_joypads() {
-	if (joy_mutex.try_lock() != OK) {
-		return;
+void JoypadLinux::joypad_events_thread_func(void *p_user) {
+	if (p_user) {
+		JoypadLinux *joy = (JoypadLinux *)p_user;
+		joy->joypad_events_thread_run();
+	}
+}
+
+void JoypadLinux::joypad_events_thread_run() {
+	while (!joypad_events_exit.is_set()) {
+		bool no_events = true;
+		for (int i = 0; i < JOYPADS_MAX; i++) {
+			MutexLock lock(joypads_mutex[i]);
+			Joypad &joypad = joypads[i];
+			if (joypad.fd == -1) {
+				continue;
+			}
+			input_event event;
+			while (read(joypad.fd, &event, sizeof(event)) > 0) {
+				no_events = false;
+				JoypadEvent joypad_event;
+				joypad_event.type = event.type;
+				joypad_event.code = event.code;
+				joypad_event.value = event.value;
+				joypad.events.push_back(joypad_event);
+			}
+			if (errno != EAGAIN) {
+				close_joypad(joypad, i);
+			}
+		}
+		if (no_events) {
+			usleep(10000); // 10ms
+		}
```

Comment:
Note: It would probably be better to use a `poll()` to detect changes. However:
1. This is how the other thread is being managed; so I used this approach for consistency.
2. `poll()` should be used on both threads, but this would the require refactoring the `Joypad` class (separating the OS file descriptors from the Godot data to avoid `mutex` sharing) and is probably best done in a separate PR as an enhancement instead of a bug fix.
3. The other thread (as well as the [`XEvent` polling](https://github.com/godotengine/godot/blob/d321858732c83f1aecfbc7e3de4c347cd509639e/platform/linuxbsd/display_server_x11.cpp#L3017)) uses a `select()` call (which is currently not used to detect changes, but [only to check whether a change has happened to prevent the next call from blocking](https://github.com/godotengine/godot/blob/0462f7cacce57b35e7f4957ba95eb5514a3ad314/platform/x11/joypad_linux.cpp#L178-L184)). These `select()` calls should be replaced with `poll()` calls too. Again, this is best done in a separate PR as an enhancement.

Reply:
The USB polling rate is 125 Hz by default. Shouldn't we use `8000` as a duration?

Reply:
That would make sense if there was a single joypad and the purpose of the delay was to wait until there was new data. In reality the delay would be less than 8000 μs, because there is the time needed to read the data. The required delay would also be indeterminate if there were multiple joypads, because they wouldn't be synchronised.

Note: The purpose of this delay is to not be busy waiting; especially when there are no joypads attached.